lib/file.coffee

```diff
@@ -67,3 +67,6 @@ class File
 
   isPathEqual: (pathToCompare) ->
     @path is pathToCompare or @realPath is pathToCompare
+
+  # Return a snapshot of the instance's filesystem properties for API consumption
+  getMetadata: -> {@symlink, @realPath, @status, @stats}
```

Comment:
So what's wrong with passing different arguments as a temporary solution until a formal API is developed? When it is, you can just use Grim to deprecate usage of the second argument so package authors know to use the more elegant approach.